File: 1813-maximum-erasure-value/maximum-erasure-value.go
func maximumUniqueSubarray(nums []int) int {
    seen := make(map[int]bool)
    currentSum, result := 0, 0
    start := 0

    for end := 0; end < len(nums); end++ {
        for seen[nums[end]] {
            seen[nums[start]] = false
            currentSum -= nums[start]
            start++
        }

        currentSum += nums[end]
        seen[nums[end]] = true
        if currentSum > result {
            result = currentSum
        }
    }

    return result
}

File: 1813-maximum-erasure-value/maximum-erasure-value.py
class Solution:
    def maximumUniqueSubarray(self, nums: List[int]) -> int:
        seen = set()
        current_sum = result = 0
        start = 0

        for end in range(len(nums)):
            while nums[end] in seen:
                seen.remove(nums[start])
                current_sum -= nums[start]
                start += 1

            current_sum += nums[end]
            seen.add(nums[end])
            result = max(result, current_sum)

        return result